packages/amplify-cli-npm/binary.ts

```diff
@@ -121,7 +121,10 @@ export class Binary {
     console.log(`Downloading release from ${getCompressedBinaryUrl()}`);
     try {
       const res = await axios({ url: getCompressedBinaryUrl(), responseType: 'stream' });
-      await pipeline(res.data, createGunzip(), this.extract());
+      // Dummy array to collect a promise from nested pipeline that extracts tar content to a file.
+      const extractPromiseCollector: Array<Promise<void>> = [];
+      await pipeline(res.data, createGunzip(), this.extract(extractPromiseCollector));
+      await extractPromiseCollector[0];
```

Comment:
why only await the first promise and not Promise.allSettled?

Reply:
if we get here there's only one promise, but `Promise.all` would do too.

Reply:
Changed. I used `Promise.all` not `Promise.allSettled` . This should throw if extracting fails.